File: Models/Endereco.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text.Json.Serialization;

namespace PessoasAPI.Models;

public class Endereco
{
    public int Id { get; set; }
    public string Rua { get; set; } = string.Empty;
    public string Cidade { get; set; } = string.Empty;
    public string Estado { get; set; } = string.Empty;

    public int? PessoaId { get; set; }
    public Pessoa? Pessoa { get; set; }
}

File: frontend/src/components/Endereco.js
import React, { useEffect, useState } from "react";

const API_URL = "http://localhost:5000";

function Enderecos() {
  const [enderecos, setEnderecos] = useState([]);

  useEffect(() => {
    fetch(`${API_URL}/enderecos`)
      .then((res) => res.json())
      .then((data) => setEnderecos(data))
      .catch((err) => console.error("Erro ao buscar endereços:", err));
  }, []);

  return (
    <div>
      <h2>Lista de Endereços</h2>
      <ul>
        {enderecos.map((e) => (
          <li key={e.id}>
            {e.rua}, {e.cidade}, {e.estado} (Pessoa ID: {e.pessoaId})
          </li>
        ))}
      </ul>
    </div>
  );
}

export default Enderecos;
